cmd/consolelogs/aws/getlogs.go

```diff
@@ -0,0 +1,157 @@
+package aws
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	awsutil "github.com/openshift/hypershift/cmd/infra/aws/util"
+	"github.com/openshift/hypershift/cmd/util"
+)
+
+type ConsoleLogOpts struct {
+	Name               string
+	Namespace          string
+	AWSCredentialsFile string
+	OutputDir          string
+}
+
+func NewCommand() *cobra.Command {
+
+	opts := &ConsoleLogOpts{
+		Name:      "example",
+		Namespace: "clusters",
+	}
+
+	cmd := &cobra.Command{
+		Use:          "aws",
+		Short:        "Get AWS machine instance console logs",
+		SilenceUsage: true,
+	}
+
+	cmd.Flags().StringVarP(&opts.Namespace, "namespace", "n", opts.Namespace, "A cluster namespace")
+	cmd.Flags().StringVar(&opts.Name, "name", opts.Name, "A cluster name")
+	cmd.Flags().StringVar(&opts.AWSCredentialsFile, "aws-creds", opts.AWSCredentialsFile, "Path to an AWS credentials file (required)")
+	cmd.Flags().StringVar(&opts.OutputDir, "output-dir", opts.OutputDir, "Directory where to place console logs (required)")
+
+	cmd.MarkFlagRequired("aws-creds")
+	cmd.MarkFlagRequired("output-dir")
```

Comment:
Nit: I would make the `Name` mandatory and not default it

Reply:
done